src/library/columncache.cpp

```diff
@@ -65,12 +69,59 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnIndexByEnum[COLUMN_CRATETRACKSTABLE_TRACKID] = fieldIndex(CRATETRACKSTABLE_TRACKID);
     m_columnIndexByEnum[COLUMN_CRATETRACKSTABLE_CRATEID] = fieldIndex(CRATETRACKSTABLE_CRATEID);
 
-    // Set up the reverse mapping, ignoring columns that don't get displayed.
-    for (int i = 0; i < NUM_COLUMNS; ++i) {
-        int index = m_columnIndexByEnum[i];
-        if (index < 0) {
-            continue;
-        }
-        m_columnByIndex[index] = static_cast<Column>(i);
-    }
+    const QString sortInt("cast(%1 as integer)");
+    const QString sortNoCase("lower(%1)");
+    const QString sortDefault("%1");
+    m_columnSortByIndex.clear();
```

Comment:
You could leave out all the columns that use sortDefault -- it would make it easier to tell which columns need special casing that way.